### PR TITLE
fix: Make split step work with ftp connector

### LINKED ISSUE #5737: Split not able to work with FTP connector
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
Split step is not able to handle data coming from FTP connector. The FTP connector's outcome is a RemoteFile object and split step is not able to handle this in order to split the data.

Expected behavior:
When FTP connector defines a collection JSON data shape the split should be able to handle that.

### PR DESCRIPTION
**Pull request by christophd**
Split step should auto convert types if necessary in order to handle Json data properly.

At the moment the known types for auto conversion are:
- InputStream (used by APIProvider, WebHook and Http connector)
- RemoteFile (used by ftp connector)

These types get converted to String representation using the Camel type conversion strategy and if this data happens to be a Json array split step splits those to its elements. Any other data types (e.g. List, Scanner) are handled by the Camel splitter by default splitting those to its elements.

Fixes #5737

### CODE REVIEW

File: app/integration/runtime/src/main/java/io/syndesis/integration/runtime/handlers/SplitStepHandler.java [hunk 1]
```diff
@@ -26,21 +27,50 @@
...
 import org.apache.camel.builder.Builder;
 import org.apache.camel.model.ProcessorDefinition;
 import org.apache.camel.spi.Language;
 import org.apache.camel.support.ExpressionAdapter;
 import org.apache.camel.util.ObjectHelper;
 
 public class SplitStepHandler implements IntegrationStepHandler {
+
+    /* Types that need conversion to String in order to perform a split operation */
+    private enum AUTO_CONVERT_TYPES {
```
Comment (zregvart) [inline app/integration/runtime/src/main/java/io/syndesis/integration/runtime/handlers/SplitStepHandler.java]:
```suggestion
    private enum AutoConvertedTypes {
```

PMD doesn't like it:
```
[INFO] PMD Failure: io.syndesis.integration.runtime.handlers.SplitStepHandler:47 Rule:ClassNamingConventions Priority:4 The enum name 'AUTO_CONVERT_TYPES' doesn't match '[A-Z][a-zA-Z0-9]*'.
```

Comment (christophd) [reply app/integration/runtime/src/main/java/io/syndesis/integration/runtime/handlers/SplitStepHandler.java]: Argh! This has been a final static field before and I forgot to rename. Thanks! Lessons learned: Run the checks before checkin even if you think it is not necessary

Comment (christophd) [reply app/integration/runtime/src/main/java/io/syndesis/integration/runtime/handlers/SplitStepHandler.java]: force pushed to this PR

**Comment by christophd** [comment]
@zregvart can you please have another look. Checks passing now. Thanks!

**Pull request merged**